From fe7a79d2233337aceefc3b4cf2b1a089602cb706 Mon Sep 17 00:00:00 2001
From: Grigory Entin <grigory.entin@gmail.com>
Date: Mon, 15 Feb 2021 23:20:08 +0100
Subject: [PATCH] Added support for Linux with workaround for no UTIs.

---
 .../Classes/OpenAPIs/URLSessionImplementations.swift       | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/BlinkOpenAPI/Classes/OpenAPIs/URLSessionImplementations.swift b/BlinkOpenAPI/Classes/OpenAPIs/URLSessionImplementations.swift
index e09a8ce..13f4dbd 100644
--- a/BlinkOpenAPI/Classes/OpenAPIs/URLSessionImplementations.swift
+++ b/BlinkOpenAPI/Classes/OpenAPIs/URLSessionImplementations.swift
@@ -5,7 +5,10 @@
 //
 
 import Foundation
-#if !os(macOS)
+#if os(Linux)
+import FoundationNetworking
+#endif
+#if !os(macOS) && !os(Linux)
 import MobileCoreServices
 #endif
 
@@ -555,11 +558,13 @@ private class FormDataEncoding: ParameterEncoding {
     func mimeType(for url: URL) -> String {
         let pathExtension = url.pathExtension
 
+        #if !os(Linux)
         if let uti = UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, pathExtension as NSString, nil)?.takeRetainedValue() {
             if let mimetype = UTTypeCopyPreferredTagWithClass(uti, kUTTagClassMIMEType)?.takeRetainedValue() {
                 return mimetype as String
             }
         }
+        #endif
         return "application/octet-stream"
     }
 
-- 
2.27.0

