From fc71657c38d343ff089323d3986a660745f56b11 Mon Sep 17 00:00:00 2001
From: Grigory Entin <grigory.entin@gmail.com>
Date: Mon, 15 Feb 2021 23:23:55 +0100
Subject: [PATCH] Added support for OpenCombine/Linux.

---
 BlinkOpenAPI/Classes/OpenAPIs/APIs/BlinkDefaultAPI.swift | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/BlinkOpenAPI/Classes/OpenAPIs/APIs/BlinkDefaultAPI.swift b/BlinkOpenAPI/Classes/OpenAPIs/APIs/BlinkDefaultAPI.swift
index baf2df4..58395e7 100644
--- a/BlinkOpenAPI/Classes/OpenAPIs/APIs/BlinkDefaultAPI.swift
+++ b/BlinkOpenAPI/Classes/OpenAPIs/APIs/BlinkDefaultAPI.swift
@@ -6,9 +6,13 @@
 //
 
 import Foundation
+#if os(Linux)
+import OpenCombine
+#else
 #if canImport(Combine)
 import Combine
 #endif
+#endif
 
 open class BlinkDefaultAPI {
     /**
-- 
2.27.0

